packages/template-phpstan-compiler/src/PHPStan/FileAnalyserProvider.php

```diff
@@ -30,8 +30,9 @@ public function provide(): FileAnalyser
         }
 
         $container = $this->derivativeContainerFactory->create([__DIR__ . '/../../config/php-parser.neon']);
-        $fileAnalyser = $container->getByType(FileAnalyser::class);
 
+        /** @var FileAnalyser $fileAnalyser */
+        $fileAnalyser = $container->getByType(FileAnalyser::class);
```

Comment:
phpstan has a nette-extension which defines the return-type here.. so the phpdoc should not be needed (or at least only for phpstorm, not phpstan)

https://github.com/phpstan/phpstan-src/blob/ec5946a4d98eb225796da072727225c23b35e25d/src/Internal/ContainerDynamicReturnTypeExtension.php

Reply:
This is for PHPStorm autocomplete. All generic types defined in @var in that you'll find in code here are for PHPStorm.